### PR TITLE
Add e2e tests for GetConfiguredNodesOnEndpoint direct method

### PR DESCRIPTION
**Pull request by monikavar**
Add e2e tests for GetConfiguredNodesOnEndpoint direct method

### CODE REVIEW

**Review by cristipogacean — APPROVED** [review]

File: e2e-tests/IIoTPlatform-E2E-Tests/Standalone/A_PublishSingleNodeStandaloneDirectMethodTestTheory.cs [hunk 1]
```diff
@@ -117,6 +126,15 @@ IIoTMultipleNodesTestContext context
...
             var requestGetConfiguredNodesOnEndpoint = nodesOnEndpoint.ToApiModel();
 
+            //Call GetConfiguredEndpoints direct method
+            responseGetConfiguredEndpoints = await TestHelper.CallMethodAsync(_iotHubClient, _iotHubPublisherDeviceName, _iotHubPublisherModuleName, new MethodParameterModel {
+                Name = TestConstants.DirectMethodNames.GetConfiguredEndpoints
+            }, _context, cts.Token).ConfigureAwait(false);
+
+            Assert.Equal((int)HttpStatusCode.OK, responseGetConfiguredEndpoints.Status);
+            configuredEndpointsResponse = _serializer.Deserialize<List<PublishNodesEndpointApiModel>>(responseGetConfiguredEndpoints.JsonPayload);
+            Assert.Equal(configuredEndpointsResponse.Count, 1);
```
Comment (karok2m) [inline e2e-tests/IIoTPlatform-E2E-Tests/Standalone/A_PublishSingleNodeStandaloneDirectMethodTestTheory.cs]: Please validate that all fields of returned endpoint are as expected.

File: e2e-tests/IIoTPlatform-E2E-Tests/Standalone/A_PublishSingleNodeStandaloneDirectMethodTestTheory.cs [hunk 2]
```diff
@@ -215,6 +244,17 @@ IIoTMultipleNodesTestContext context
...
+            //Call GetConfiguredEndpoints direct method
+            responseGetConfiguredEndpoints = await TestHelper.CallMethodAsync(_iotHubClient, _iotHubPublisherDeviceName, _iotHubPublisherModuleName, new MethodParameterModel {
+                Name = TestConstants.DirectMethodLegacyNames.GetConfiguredEndpoints
+            }, _context, cts.Token).ConfigureAwait(false);
+
+            Assert.Equal((int)HttpStatusCode.OK, responseGetConfiguredEndpoints.Status);
+            epObj = JObject.Parse(responseGetConfiguredEndpoints.JsonPayload);
+            endpoints = _serializer.SerializeToString(epObj["Endpoints"]);
+            configuredEndpointsResponse = _serializer.Deserialize<List<PublishNodesEndpointApiModel>>(endpoints);
+            Assert.Equal(configuredEndpointsResponse.Count, 1);
```
Comment (karok2m) [inline e2e-tests/IIoTPlatform-E2E-Tests/Standalone/A_PublishSingleNodeStandaloneDirectMethodTestTheory.cs:257]: Please validate that all fields of returned endpoint are as expected.

File: e2e-tests/IIoTPlatform-E2E-Tests/Standalone/B_PublishMultipleNodesStandaloneDirectMethodTestTheory.cs [hunk 3]
```diff
@@ -114,6 +123,15 @@ IIoTMultipleNodesTestContext context
...
             await Task.Delay(TestConstants.DefaultTimeoutInMilliseconds, cts.Token);
 
+            //Call GetConfiguredEndpoints direct method
+            responseGetConfiguredEndpoints = await TestHelper.CallMethodAsync(_iotHubClient, _iotHubPublisherDeviceName, _iotHubPublisherModuleName, new MethodParameterModel {
+                Name = TestConstants.DirectMethodNames.GetConfiguredEndpoints
+            }, _context, cts.Token).ConfigureAwait(false);
+
+            Assert.Equal((int)HttpStatusCode.OK, responseGetConfiguredEndpoints.Status);
+            configuredEndpointsResponse = _serializer.Deserialize<List<PublishNodesEndpointApiModel>>(responseGetConfiguredEndpoints.JsonPayload);
+            Assert.Equal(configuredEndpointsResponse.Count, 1);
```
Comment (karok2m) [inline e2e-tests/IIoTPlatform-E2E-Tests/Standalone/B_PublishMultipleNodesStandaloneDirectMethodTestTheory.cs]: Please validate that all fields of the returned endpoint are as expected.

File: e2e-tests/IIoTPlatform-E2E-Tests/Standalone/B_PublishMultipleNodesStandaloneDirectMethodTestTheory.cs [hunk 4]
```diff
@@ -229,6 +258,17 @@ IIoTMultipleNodesTestContext context
...
+            //Call GetConfiguredEndpoints direct method
+            responseGetConfiguredEndpoints = await TestHelper.CallMethodAsync(_iotHubClient, _iotHubPublisherDeviceName, _iotHubPublisherModuleName, new MethodParameterModel {
+                Name = TestConstants.DirectMethodLegacyNames.GetConfiguredEndpoints
+            }, _context, cts.Token).ConfigureAwait(false);
+
+            Assert.Equal((int)HttpStatusCode.OK, responseGetConfiguredEndpoints.Status);
+            epObj = JObject.Parse(responseGetConfiguredEndpoints.JsonPayload);
+            endpoints = _serializer.SerializeToString(epObj["Endpoints"]);
+            configuredEndpointsResponse = _serializer.Deserialize<List<PublishNodesEndpointApiModel>>(endpoints);
+            Assert.Equal(configuredEndpointsResponse.Count, 1);
```
Comment (karok2m) [inline e2e-tests/IIoTPlatform-E2E-Tests/Standalone/B_PublishMultipleNodesStandaloneDirectMethodTestTheory.cs]: Please validate that all fields of the returned endpoint are as expected.

File: e2e-tests/IIoTPlatform-E2E-Tests/Standalone/C_PublishMultipleEndpointsStandaloneDirectMethodTestTheory.cs [hunk 5]
```diff
@@ -126,6 +135,15 @@ IIoTMultipleNodesTestContext context
...
             await Task.Delay(TestConstants.DefaultTimeoutInMilliseconds, cts.Token);
 
+            //Call GetConfiguredEndpoints direct method
+            responseGetConfiguredEndpoints = await TestHelper.CallMethodAsync(_iotHubClient, _iotHubPublisherDeviceName, _iotHubPublisherModuleName, new MethodParameterModel {
+                Name = TestConstants.DirectMethodNames.GetConfiguredEndpoints
+            }, _context, cts.Token).ConfigureAwait(false);
+
+            Assert.Equal((int)HttpStatusCode.OK, responseGetConfiguredEndpoints.Status);
+            configuredEndpointsResponse = _serializer.Deserialize<List<PublishNodesEndpointApiModel>>(responseGetConfiguredEndpoints.JsonPayload);
+            Assert.Equal(configuredEndpointsResponse.Count, 2);
```
Comment (karok2m) [inline e2e-tests/IIoTPlatform-E2E-Tests/Standalone/C_PublishMultipleEndpointsStandaloneDirectMethodTestTheory.cs:145]: Please validate that all fields of both returned endpoints are as expected.

File: e2e-tests/IIoTPlatform-E2E-Tests/Standalone/C_PublishMultipleEndpointsStandaloneDirectMethodTestTheory.cs [hunk 6]
```diff
@@ -257,6 +295,17 @@ IIoTMultipleNodesTestContext context
...
+            //Call GetConfiguredEndpoints direct method
+            responseGetConfiguredEndpoints = await TestHelper.CallMethodAsync(_iotHubClient, _iotHubPublisherDeviceName, _iotHubPublisherModuleName, new MethodParameterModel {
+                Name = TestConstants.DirectMethodLegacyNames.GetConfiguredEndpoints
+            }, _context, cts.Token).ConfigureAwait(false);
+
+            Assert.Equal((int)HttpStatusCode.OK, responseGetConfiguredEndpoints.Status);
+            epObj = JObject.Parse(responseGetConfiguredEndpoints.JsonPayload);
+            endpoints = _serializer.SerializeToString(epObj["Endpoints"]);
+            configuredEndpointsResponse = _serializer.Deserialize<List<PublishNodesEndpointApiModel>>(endpoints);
+            Assert.Equal(configuredEndpointsResponse.Count, 2);
```
Comment (karok2m) [inline e2e-tests/IIoTPlatform-E2E-Tests/Standalone/C_PublishMultipleEndpointsStandaloneDirectMethodTestTheory.cs]: Please validate that all fields of both returned endpoints are as expected.

File: e2e-tests/IIoTPlatform-E2E-Tests/Standalone/A_PublishSingleNodeStandaloneDirectMethodTestTheory.cs (hunk 1 above)
Comment (monikavar) [reply e2e-tests/IIoTPlatform-E2E-Tests/Standalone/A_PublishSingleNodeStandaloneDirectMethodTestTheory.cs]: The sample request has only EndpointUrl; all other fields are null. Will check accordingly.

**Pull request merged**